x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/ApiKeyFieldNameTranslators.java

```diff
@@ -38,37 +61,243 @@ public class ApiKeyFieldNameTranslators {
+        } else if (queryBuilder instanceof final SimpleQueryStringBuilder query) {
+            SimpleQueryStringBuilder simpleQueryStringBuilder = QueryBuilders.simpleQueryStringQuery(query.value());
+            Map<String, Float> queryFields = new HashMap<>(query.fields());
+            // be explicit that no field means all fields
+            if (queryFields.isEmpty()) {
+                queryFields.put("*", AbstractQueryBuilder.DEFAULT_BOOST);
+            }
+            // override "lenient" if querying all the fields, because, due to different field mappings,
+            // the query parsing will almost certainly fail otherwise
+            if (QueryParserHelper.hasAllFieldsWildcard(queryFields.keySet())) {
+                simpleQueryStringBuilder.lenient(true);
+            } else {
+                simpleQueryStringBuilder.lenient(query.lenient());
+            }
+            // translate query-level field name patterns to index-level concrete field names
+            for (Map.Entry<String, Float> requestedFieldNameOrPattern : queryFields.entrySet()) {
+                for (String translatedField : translatePattern(requestedFieldNameOrPattern.getKey())) {
+                    simpleQueryStringBuilder.fields()
+                        .compute(
+                            translatedField,
+                            (k, v) -> (v == null) ? requestedFieldNameOrPattern.getValue() : v * requestedFieldNameOrPattern.getValue()
+                        );
+                    fieldNameVisitor.accept(translatedField);
+                }
+            }
+            if (simpleQueryStringBuilder.fields().isEmpty()) {
+                // A SimpleQueryStringBuilder with empty fields() will eventually produce a SimpleQueryString
+                // Lucene query that accesses all the fields, including disallowed ones.
+                // Instead, the behavior we're after here is that a query that accesses only disallowed fields
+                // mustn't match any docs.
+                return new MatchNoneQueryBuilder().boost(simpleQueryStringBuilder.boost()).queryName(simpleQueryStringBuilder.queryName());
+            }
+            return simpleQueryStringBuilder.analyzer(query.analyzer())
+                .defaultOperator(query.defaultOperator())
+                .minimumShouldMatch(query.minimumShouldMatch())
+                .flags(query.flags())
+                .type(query.type())
+                .quoteFieldSuffix(query.quoteFieldSuffix())
+                .analyzeWildcard(query.analyzeWildcard())
+                .autoGenerateSynonymsPhraseQuery(query.autoGenerateSynonymsPhraseQuery())
+                .fuzzyTranspositions(query.fuzzyTranspositions())
+                .fuzzyMaxExpansions(query.fuzzyMaxExpansions())
+                .fuzzyPrefixLength(query.fuzzyPrefixLength())
+                .boost(query.boost())
+                .queryName(query.queryName());
+        } else {
```

Comment:
This is the important change in this PR. It copies, rather than in-place modifies, the `SimpleQueryStringBuilder`.